targets/TARGET_STM/rtc_api.c

```diff
@@ -144,28 +144,8 @@ void rtc_init(void)
 
 void rtc_free(void)
 {
-#if !MBED_CONF_TARGET_LSE_AVAILABLE
```

Comment:
I can't find info for this removal in the commit message

Can you provide description how is this being updated? `MBED_CONF_TARGET_LSE_AVAILABLE` why is being removed, and adding `TARGET_STM32F1` - are these 2 related?